lib/performance/EmittedAssetSizeLimitPlugin.js

```diff
@@ -0,0 +1,144 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Sean Larkin @thelarkinn
+*/
+var path = require('path');
+
+var EntrypointsOverSizeLimitWarning = require('./EntrypointsOverSizeLimitWarning');
+var AssetsOverSizeLimitWarning = require('./AssetsOverSizeLimitWarning');
+var NoAsyncChunksWarning = require('./NoAsyncChunksWarning');
+
+function EmittedAssetSizeLimitPlugin(performanceOptions) {
+	this.maxAssetSize = performanceOptions.maxAssetSize;
+	this.maxInitialSize = performanceOptions.maxInitialChunkSize;
+	this.hints = performanceOptions.hints;
+	this.errorOnHint = performanceOptions.errorOnHint;
+}
+
+module.exports = EmittedAssetSizeLimitPlugin;
+
+// When using this we should always
+// compare byte size and then format later
+function doesExceedLimit(limit, actualSize) {
+	return limit < actualSize;
+}
+
+function isAssetJsFile(assetFilename) {
+	var jsRegex = /\.js($|\?)/i;
+
+	return jsRegex.test(assetFilename);
+}
+
+EmittedAssetSizeLimitPlugin.prototype.apply = function(compiler) {
+	if(!this.hints) {
+		return;
+	}
+	var entrypointSizeLimit = this.maxInitialSize;
+	var sizeLimit = this.maxAssetSize;
+	var hints = this.hints;
+	var shouldErrorOnHint = this.errorOnHint;
+
+	compiler.plugin("emit", function(compilation, callback) {
```

Comment:
`after-emit` is the better hook here. In `emit` it's still allowed to add more assets to the compilation.